clang/lib/Sema/SemaExpr.cpp

```diff
@@ -13367,6 +13367,21 @@ static void DiagnoseConstAssignment(Sema &S, const Expr *E,
         if (!DiagnosticEmitted) {
           S.Diag(Loc, diag::err_typecheck_assign_const)
               << ExprRange << ConstVariable << VD << VD->getType();
+          ExprResult Deref;
+          Expr *TE = const_cast<Expr *>(E);
```

Comment:
I think it would be more prone to modifications due to it being non `const` and being used in many other contexts, maybe we could leave it as it is for now?

Reply:
If it’s easier to use `const_cast` here than to change the parameter to be non-`const`, then that’s fine imo. Our const-correctness w/ AST nodes is all over the place anyway, so while not ‘ideal’, it’s not the end of the world imo